### PR TITLE
Document all class parameters

### PR DESCRIPTION
**Pull request by ekohl**
Closes GH-71

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -3,17 +3,91 @@
...
+#                          type:Stdlib::Absolutepath
+#
+# $localzonepath::         File holding local zones like RFC1912 or RFC1918 files.
+#                          type:Optional[Stdlib::Absolutepath]
+#
+# $forward::               The forward option
+#                          type:Optional[String]
+#
+# $forwarders::            The forwarders option
+#                          type:Array[String]
```
Comment (stbenjam) [inline manifests/init.pp]: I know we have a bunch of cases where we're expecting IP's, maybe we should get some better validation on them?

This looks promising: https://github.com/thrnio/puppet-ip

File: manifests/init.pp [hunk 2]
```diff
@@ -3,17 +3,91 @@
...
+#
+# $listen_on_v6::          The listen-on-v6 option
+#                          type:Optional[String]
+#
+# $recursion::             The recursion option
+#                          type:Enum[yes, no]
```
Comment (stbenjam) [inline manifests/init.pp]: I feel like these should be booleans that the templates convert to yes/no strings.

File: manifests/init.pp (hunk 1 above)
Comment (ekohl) [inline manifests/init.pp]: I think we could use ```Stdlib::Compat::Ipv4``` and ```Stdlib::Compat::Ipv6``` as well.

Comment (ekohl) [inline manifests/init.pp]: According to http://www.zytrax.com/books/dns/ch7/queries.html#forwarders they can also include a port so simply forcing it to an IP is not the correct thing to do.

Comment (stbenjam) [inline manifests/init.pp]: I suppose it's fine as is then, unless you wanted to create a type for DNS forwarders.

Comment (ekohl) [inline manifests/init.pp]: Given this class isn't exposed in the installer that currently adds nothing.

Comment (stbenjam) [inline manifests/init.pp]: It would once the variables themselves are typed, but I guess we can cross that bridge when we drop puppet 3.

Comment (ekohl) [inline manifests/init.pp]: Exactly

**Comment by mmoll** [comment]
merged, thanks @stbenjam  and @ekohl!

**Comment by ekohl** [comment]
Lint now fails with the following error:
```
manifests/init.pp:126:WARNING: missing documentation for class parameter dns::additional_directives
```

I'll come up with a patch but the gate at the airport just opened.

**Pull request merged**